research/object_detection/meta_architectures/detr_lib.py

```diff
@@ -0,0 +1,151 @@
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Defines libraries for the Detection Transformer model in TF 2.0.
+
+Model paper: https://arxiv.org/abs/2005.12872
+Transformer model code source: https://github.com/tensorflow/tensor2tensor
+"""
+import tensorflow as tf
+from object_detection.utils import shape_utils
+from official.nlp.modeling import layers
+from official.modeling import tf_utils
+
+import math
+
+import tensorflow as tf
+
+
+class Attention(tf.keras.layers.Layer):
```

Comment:
this guy is mostly copied from official/nlp/transformer and @xinliupitt has done some work to make sure this can be fully replaced by layers.MultiHeadAttention

Reply:
sounds good

Reply:
As discussed offline, we will update this to use Kera's version when it lands.